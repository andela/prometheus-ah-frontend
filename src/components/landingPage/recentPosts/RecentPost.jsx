import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';

const RecentPost = ({
  image, title, date, category, readTime, author
}) => {
  const formatedDate = moment(date).startOf('hour').fromNow();
  return (
    <div className="card shadow-lg border-0 mb-5 recent-post">
      <img src={image} className="img-fluid recent-post-image" alt="" />
      <div className="card-body">
        <h3 className="card-title post-title">{title}</h3>
        <div className="d-flex justify-content-between">
          <div className="card-text post-author">
            <p className="author">{author}</p>
            <p><small className="text-muted font-italic">{readTime}</small></p>
          </div>
          <div className="card-text post-time">
            <p className="font-weight-light font-italic">{category}</p>
            <p className="time">{formatedDate}</p>
          </div>
        </div>
      </div>
    </div>
  );
};

RecentPost.propTypes = {
  image: PropTypes.string,
  title: PropTypes.string,
  date: PropTypes.string,
  category: PropTypes.string,
  readTime: PropTypes.string,
  author: PropTypes.string
};

export default RecentPost;
