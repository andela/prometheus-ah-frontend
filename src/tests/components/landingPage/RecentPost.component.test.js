import React from 'react';
import { shallow } from 'enzyme';
import RecentPost from '../../../components/landingPage/recentPosts/RecentPost';
import mockData from '../../__mocks__/mockData';

let wrapper;

describe('RecentPost component', () => {
  beforeEach(() => {
    const {
      image, title, date, category, readTime,
    } = mockData.postProps;
    const props = {
      image,
      title,
      date,
      category,
      readTime,
      author: {
        username: 'faksam',
        firstname: null,
        lastname: null
      }
    };

    wrapper = shallow(<RecentPost {...props} />);
  });

  test('should render a snapshot of the RecentPost component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
