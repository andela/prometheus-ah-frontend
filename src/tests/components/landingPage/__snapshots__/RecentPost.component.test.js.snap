// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentPost component should render a snapshot of the RecentPost component 1`] = `
<div
  className="card shadow-lg border-0 mb-5 recent-post"
>
  <img
    alt=""
    className="img-fluid recent-post-image"
    src="https://image.ibb.co/eLVQwL/kids.jpg"
  />
  <div
    className="card-body"
  >
    <Link
      className="card-title post-title"
      replace={false}
      to="/articles/undefined"
    >
      My Broken Mother, My Broken Heart
    </Link>
    <div
      className="d-flex justify-content-between mt-3"
    >
      <div
        className="card-text post-author"
      >
        <p
          className="author"
        >
          faksam
        </p>
        <p>
          <small
            className="text-muted font-italic"
          >
            1 min read
          </small>
        </p>
      </div>
      <div
        className="card-text post-time"
      >
        <p
          className="font-weight-light font-italic text-uppercase"
        >
          Technology
        </p>
        <p
          className="time"
        >
          a month ago
        </p>
      </div>
    </div>
  </div>
</div>
`;
